patroni/postgresql/config.py

```diff
@@ -1007,16 +1013,20 @@ def reload_config(self, config, sighup=False):
         else:
             logger.info('No PostgreSQL configuration items changed, nothing to reload.')
 
-    def set_synchronous_standby(self, name):
+    def set_synchronous_standby(self, sync_members):
         """Sets a node to be synchronous standby and if changed does a reload for PostgreSQL."""
-        if name and name != '*':
-            name = quote_ident(name)
-        if name != self._synchronous_standby_names:
-            if name is None:
+        if sync_members and set(sync_members) != set(['*']):
+            sync_members = [quote_ident(x) for x in sync_members]
+        if self._postgresql.major_version >= 90600:
+            sync_param = sync_members and '{} ({})'.format(len(sync_members), ','.join(sync_members)) or None
```

Comment:
The automatic reduction of `synchronous_node_count` is fine only for non-strict synchronous mode.

Let's say we have `synchronous_mode_strict = True`, `synchronous_node_count = 2` and `sync_members = ['*']`.
In this case we want to put `2 (*)` into `synchronous_standby_names`.

Reply:
Intention was to make the numsync dynamically adjusted by Patroni so it won't affect leader as replica nodes join and leave. We could add additional parameter to keep min_synchronous_node_count and act accordingly?

Reply:
Hmm, probably you are right. So far semantic of `synchronous_mode_strict` was that we require at least one sync node and suggested implementation doesn't change it.
Ok, let's leave it as it is for now, and later we could agree on configuration options names, i.e. whether we should use `synchronous_node_count` and `synchronous_node_count` or maybe better `replication_factor` and `minimum_replication_factor`, like it was proposed in the #672.

Reply:
Agree, It make sense to have configurable parameter for minimum synchronous stanbys instead of hard limit of 1. I will update the code accordingly.